ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/block/BlockProcessorCapella.java

```diff
@@ -55,4 +77,76 @@ public BlockProcessorCapella(
+      final BeaconState preState,
+      final SignedBeaconBlock block,
+      final BeaconState postState,
+      final IndexedAttestationCache indexedAttestationCache,
+      final BLSSignatureVerifier signatureVerifier) {
+    return BlockValidationResult.allOf(
+        () ->
+            super.validateBlock(
+                preState, block, postState, indexedAttestationCache, signatureVerifier),
+        () ->
+            verifyBlsToExecutionChanges(
+                BeaconStateCapella.required(preState),
+                BeaconBlockBodyCapella.required(block.getMessage().getBody())
+                    .getBlsToExecutionChanges(),
+                signatureVerifier));
+  }
+
+  @Override
+  protected void processOperationsNoValidation(
+      final MutableBeaconState state,
+      final BeaconBlockBody body,
+      final IndexedAttestationCache indexedAttestationCache)
+      throws BlockProcessingException {
+    super.processOperationsNoValidation(state, body, indexedAttestationCache);
+
+    processBlsToExecutionChangesNoValidation(
+        MutableBeaconStateCapella.required(state),
+        BeaconBlockBodyCapella.required(body).getBlsToExecutionChanges());
+  }
+  // process_bls_to_execution_change
+  public void processBlsToExecutionChangesNoValidation(
+      final MutableBeaconStateCapella state,
+      final SszList<SignedBlsToExecutionChange> signedBlsToExecutionChanges) {
+
+    for (SignedBlsToExecutionChange signedBlsToExecutionChange : signedBlsToExecutionChanges) {
+      BlsToExecutionChange addressChange = signedBlsToExecutionChange.getMessage();
+      final int validatorIndex = addressChange.getValidatorIndex().intValue();
+      Validator validator = state.getValidators().get(validatorIndex);
+      state
+          .getValidators()
+          .set(
+              validatorIndex,
+              validator.withWithdrawalCredentials(
+                  withdrawalAddressFromEth1Address(addressChange.getToExecutionAddress())));
+    }
+  }
+
+  public static Bytes32 withdrawalAddressFromEth1Address(final Bytes20 toExecutionAddress) {
+    return Bytes32.wrap(
+        Bytes.concatenate(
+            ETH1_WITHDRAWAL_PREFIX, WITHDRAWAL_BUFFER, toExecutionAddress.getWrappedBytes()));
+  }
+
+  private BlockValidationResult verifyBlsToExecutionChanges(
+      final BeaconState genericState,
+      final SszList<SignedBlsToExecutionChange> signedBlsToExecutionChanges,
+      final BLSSignatureVerifier signatureVerifier) {
+    for (SignedBlsToExecutionChange signedBlsToExecutionChange : signedBlsToExecutionChanges) {
+      boolean signatureValid =
+          operationSignatureVerifier.verifyBlsToExecutionChangeSignature(
+              genericState.getFork(), genericState, signedBlsToExecutionChange, signatureVerifier);
+      if (!signatureValid) {
+        return BlockValidationResult.failed(
+            "BlsToExecutionChange signature is invalid: " + signedBlsToExecutionChange);
```

Comment:
Worth noting that this failure reason is currently incorrect - it's being used for all the validation failures that could happen, not just invalid signature.